snuba/utils/streams/consumers/backends/abstract.py

```diff
@@ -94,6 +94,24 @@ def seek(self, offsets: Mapping[TStream, TOffset]) -> None:
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def pause(self, streams: Sequence[TStream]) -> None:
+        """
+        Pause the consumption of messages for the provided streams.
+
+        Raises a ``RuntimeError`` if called on a closed consumer.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def resume(self, streams: Sequence[TStream]) -> None:
+        """
+        Resume the consumption of messages for the provided streams.
+
+        Raises a ``RuntimeError`` if called on a closed consumer.
```

Comment:
> Should we raise also if we try to resume a consumer that is not paused ? It is still an invalid state.

Since resuming a consumer that is not paused doesn't have any unexpected side effects (resuming N times is no different than resuming a single time) I don't think it is necessary to raise here. I'm maybe okay with logging a warning here (but even that seems a little heavy handed.)

Basically, it seems potentially symptomatic of incorrect usage to me, but the specific use of this method is not going to materially change the state of the consumer so I don't think we need to be that concerned.

Reply:
mh, ok if resuming a running consumer does not . compromise the state, confluent does not complain, and the consumer remains in a working and consistent state I have no issue with this.